packages/grid_client/scripts/mass_deployments.ts

```diff
@@ -1,21 +1,116 @@
+  await grid3.machines.twinDeploymentHandler.checkNodesCapacity(twinDeployments);
+  await grid3.machines.twinDeploymentHandler.checkFarmIps(twinDeployments);
+
+  const contracts: DeploymentResultContracts = {
+    created: [],
+    updated: [],
+    deleted: [],
+  };
+
+  const resultContracts: DeploymentResultContracts = {
+    created: [],
+    updated: [],
+    deleted: [],
+  };
+
+  const successfulNodesSet: number[] = [];
+  const failedNodesSet: number[] = [];
+  const original_deployments: Deployment[] = [];
+
+  const deploymentPromises: Promise<void>[] = [];
+
+  for (const twinDeployment of twinDeployments) {
+    deploymentPromises.push(
+      (async () => {
+        try {
+          const extrinsics = await grid3.machines.twinDeploymentHandler.PrepareExtrinsic(twinDeployment, contracts);
+          const extrinsicResults: Contract[] =
+            await grid3.machines.twinDeploymentHandler.tfclient.applyAllExtrinsics<Contract>([
+              ...extrinsics.nodeExtrinsics,
+              ...extrinsics.nameExtrinsics,
+            ]);
+
+          for (const contract of extrinsicResults) {
+            const updatedContract = contracts.updated.filter(c => c["contractId"] === contract.contractId);
+            if (updatedContract.length === 0) contracts.created.push(contract);
+          }
+
+          if (twinDeployment.operation === Operations.deploy) {
+            events.emit("logs", `Sending deployment to node_id: ${twinDeployment.nodeId}`);
+            for (const contract of extrinsicResults) {
+              if (twinDeployment.deployment.challenge_hash() === contract.contractType.nodeContract.deploymentHash) {
+                twinDeployment.deployment.contract_id = contract.contractId;
+                resultContracts.created.push(contract);
+                break;
+              }
+            }
+            await grid3.machines.twinDeploymentHandler.sendToNode(twinDeployment);
+            successfulNodesSet.push(twinDeployment.nodeId);
+          } else if (twinDeployment.operation === Operations.update) {
+            const old_contract_id = twinDeployment.deployment.contract_id;
+            if (old_contract_id) {
+              original_deployments.push(
+                await grid3.machines.twinDeploymentHandler.getDeploymentFromFactory(old_contract_id),
+              );
+            }
+            events.emit("logs", `Updating deployment with contract_id: ${twinDeployment.deployment.contract_id}`);
+            for (const contract of extrinsicResults) {
+              if (twinDeployment.deployment.challenge_hash() === contract.contractType.nodeContract.deploymentHash) {
+                twinDeployment.nodeId = contract.contractType.nodeContract.nodeId;
+                resultContracts.updated.push(contract);
+                break;
+              }
+            }
+            await grid3.machines.twinDeploymentHandler.sendToNode(twinDeployment);
+            successfulNodesSet.push(twinDeployment.nodeId);
```

Comment:
after sending the deployment to the node, we should wait and check the deployment result